clang/lib/Frontend/CompilerInstance.cpp

```diff
@@ -881,15 +846,6 @@ llvm::cas::CASDB &CompilerInstance::getOrCreateCAS() {
   if (CAS)
     return *CAS;
 
-  // Return the CAS from the VFS, if any.
-  //
-  // FIXME: This layering is hard to follow; instead, change clients to set the
-  // CAS when building a VFS with a CAS not available from CompilerInvocation.
-  if (FileMgr)
-    if (llvm::cas::CASDB *FSCAS =
-            extractCASFromFS(FileMgr->getVirtualFileSystem()))
-      return *FSCAS;
-
```

Comment:
I know I wrote this twisted code, but I'm having some trouble figuring out exactly why, or why it's safe to delete it all now.

Can you explain why it's okay?

(Maybe it predates getOrCreateCAS() existing, and should have been deleted then?)

Reply:
Maybe it was more useful before the consolidation of `CASOptions`? It is easier now for clients to setup a CAS and pass it along.

Reply:
Looking at git-blame, this was all added at the same time that `CASOptions` was consolidated: 112f361cfb244a209a38f5f2bdb53e4a1b818331.

Reply:
I've convinced myself that the patch is correct. The FIXME I just added a comment to in the deleted code in `setFileManager()` was apparently a future which this patch is executing :).